### PR TITLE
Keep dominant speakers on the first page of overflow gallery

### PR DESCRIPTION
**Pull request by mgamis-msft**
# What

Using a callback to listen for the children per page of the overflow gallery and updating the ordering algorithm of horizontal gallery 

# Why

https://skype.visualstudio.com/SPOOL/_workitems/edit/3138544
https://skype.visualstudio.com/SPOOL/_workitems/edit/3124011

# How Tested

Using storybook. Will verify more if bug bash.

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-components/src/components/VideoGallery/OverflowGallery.tsx [hunk 1]
```diff
@@ -99,6 +102,9 @@ export const OverflowGallery = (props: {
 
   /* @conditional-compile-remove(pinned-participants) */
   if (isNarrow) {
+    // There are no pages for ScrollableHorizontalGallery so we will approximate the first 3 remote
+    // participant tiles are visible
+    onChildrenPerPageChange?.(3);
```
Comment (dmceachernmsft) [inline packages/react-components/src/components/VideoGallery/OverflowGallery.tsx:107]: is this with the assumption that the first 3 spots will be the ones that are being swapped around?

File: packages/react-components/src/components/VideoGallery/utils/videoGalleryLayoutUtils.ts [hunk 2]
```diff
@@ -32,42 +32,46 @@ const DEFAULT_MAX_REMOTE_VIDEOSTREAMS = 4;
...
+  const visibleHorizontalGalleryParticipants = useRef<VideoGalleryRemoteParticipant[]>([]);
 
   const {
-    remoteParticipants,
+    remoteParticipants = [],
     dominantSpeakers,
     maxRemoteVideoStreams = DEFAULT_MAX_REMOTE_VIDEOSTREAMS,
-    maxAudioDominantSpeakers = DEFAULT_MAX_AUDIO_DOMINANT_SPEAKERS,
-    isScreenShareActive = false
+    maxHorizontalGalleryDominantSpeakers: maxAudioDominantSpeakers = DEFAULT_MAX_AUDIO_DOMINANT_SPEAKERS,
```
Comment (dmceachernmsft) [inline packages/react-components/src/components/VideoGallery/utils/videoGalleryLayoutUtils.ts:42]: should this be `maxOverflowGalleryDominantSpeakers`?

File: packages/react-components/src/components/VideoGallery/utils/videoGalleryLayoutUtils.ts [hunk 3]
```diff
@@ -76,12 +80,12 @@ const _useOrganizedParticipants = (props: OrganizedParticipantsArgs): OrganizedP
...
     /* @conditional-compile-remove(PSTN-calls) */ /* @conditional-compile-remove(one-to-n-calling) */
-    return visibleVideoParticipants.current.length > 0
-      ? visibleVideoParticipants.current
-      : visibleAudioParticipants.current.concat(callingParticipants);
-    return visibleVideoParticipants.current.length > 0
-      ? visibleVideoParticipants.current
-      : visibleAudioParticipants.current;
+    return visibleGridParticipants.current.length > 0
+      ? visibleGridParticipants.current
+      : visibleHorizontalGalleryParticipants.current.concat(callingParticipants);
```
Comment (dmceachernmsft) [inline packages/react-components/src/components/VideoGallery/utils/videoGalleryLayoutUtils.ts:85]: Same here i think this should be `overflow` participants?

**Review by dmceachernmsft — APPROVED** [review]
Overall looks good just a couple of naming things! 😊

File: packages/react-components/src/components/VideoGallery/utils/videoGalleryLayoutUtils.ts (hunk 3 above)
Comment (mgamis-msft) [reply packages/react-components/src/components/VideoGallery/utils/videoGalleryLayoutUtils.ts:87]: Good point. Let me do that in refactor PR

File: packages/react-components/src/components/VideoGallery/utils/videoGalleryLayoutUtils.ts (hunk 2 above)
Comment (mgamis-msft) [reply packages/react-components/src/components/VideoGallery/utils/videoGalleryLayoutUtils.ts]: Yup. Thanks.

File: packages/react-components/src/components/VideoGallery/OverflowGallery.tsx (hunk 1 above)
Comment (mgamis-msft) [reply packages/react-components/src/components/VideoGallery/OverflowGallery.tsx:107]: Yes

**Review by edwardlee-msft — APPROVED** [review]
The code looks fine to me from an initial look. 
I'm not able to deep dive due to other work, so you can ask for another reviewers opinion but overall this looks good to me!

**Pull request merged**